addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -1974,6 +1975,21 @@ var SnippetsMenu = Widget.extend({
             return createInvisibleElements(rootInvisibleSnippetEls, false, $invisibleDOMPanelEl[0]);
         }, false);
     },
+    /**
+     * Updates the "Invisible Elements" panel and shows the snippets that have a
+     * conditional visibility.
+     *
+     * @private
+     */
+    _onUpdateInvisibleElements: function () {
+        const conditionalVisibilityElEditors = this.snippetEditors.filter(editor => {
+            return editor.$target[0].dataset.visibility === "conditional";
+        });
+        for (const conditionalVisibilityElEditor of conditionalVisibilityElEditors) {
+            conditionalVisibilityElEditor.toggleTargetVisibility(true);
+        }
```

Comment:
You do one loop over `this.snippetEditors` to construct `conditionalVisibilityElEditors` then you loop over `conditionalVisibilityElEditors` to `toggleTargetVisibility`. Can't you do it in one loop ?
```
for editor of this.snippetEditors {
    if (editor.$target[0].dataset.visibility === "conditional") {
        editor.toggleTargetVisibility(true)
```